students/jemielity_kamil/lesson_13_objects_and_classes/selective_shallow_compare.py

```diff
@@ -0,0 +1,26 @@
+
+class Object1:
+    name = 'John'
+    surname = 'Travis'
+    age = 10
+
+
+class Object2:
+    name = 'John'
+    surname = 'Scott'
+    age = 32
+
+
+def shallow_compare(obj1, obj2, attr_list):
+    for element in attr_list:
+        if getattr(obj1, element) == getattr(obj2, element):
+            print('Values of attr ' + element + ' are the same')
+        else:
+            print('Values of attr ' + element + ' are different')
+
+
+object1 = Object1
+object2 = Object2
```

Comment:
Nie jestem pewien czy wiesz co dokładnie tutaj zrobiłeś, więc zerknijmy na to:
```
> print(Object1) # printuję klasę
<class '__main__.Object1'>
> print(object1) # printuję obiekt
<class '__main__.Object1'> # <- odpowiedź jest taka sama
```
W pythonie klasy też są obiektami, dla których można zrobic introspekcję i sprawdzić jakie mają atrybuty. Dzięki temu Twój program działa. Zdefiniowałeś swoją klasę w ten sposób:
```
class Object1:
    name = 'John'
    surname = 'Travis'
    age = 10
```
Taka definicja tworzy atrybuty klas, a nie atrybuty obiektów, które potem na podstawie tych klas są tworzone: https://docs.python.org/3/tutorial/classes.html#class-objects
Ostatecznie przypisałeś klasy do innych zmiennych poprzez
```
object1 = Object1
object2 = Object2
```
To nie tworzy nowych obiektów, tylko przypisuje klasę do nowej nazwy.

Podejrzewam, że to co prawdopodobnie chciałeś zrobić to utworzyć dwie różne klasy. Utworzyć z nich dwa różne obiekty i porównać parametry przez `shallow_compare` czyli coś takiego:
```
class Person1:
  def __init__(self, name, surname, age, weight):
    self.name = name
    self.surname = surname
    self.age = age
    self.weight = weight

class Person2:
  def __init__(self, name, surname, age):
    self.name = name
    self.surname = surname
    self.age = age


person1 = Person1('John', 'Travis', 10, 100)
person2 = Person2('John', 'Scott', 32)

shallow_compare(person1, person2, ['name', 'surname', 'age'])
```